### PR TITLE
add climaparams to update_aux

### PR DESCRIPTION
**Pull request by costachris**
Use parameter boxes in `update_aux.jl` and propagate down where needed

### CODE REVIEW

File: src/closures/entr_detr.jl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #### Entrainment-Detrainment kernels
 
 function compute_turbulent_entrainment(param_set, a_up::FT, w_up::FT, tke::FT, H_up::FT) where {FT}
```
Comment (odunbar) [inline src/closures/entr_detr.jl:3]: `param_set` type here?

File: src/closures/entr_detr.jl [hunk 2]
```diff
@@ -15,7 +15,7 @@ end
 function compute_inverse_timescale(param_set, b_up::FT, b_en::FT, w_up::FT, w_en::FT, tke::FT) where {FT}
```
Comment (odunbar) [inline src/closures/entr_detr.jl:15]: `param_set`  type here?

File: src/closures/entr_detr.jl [hunk 3]
```diff
@@ -25,7 +25,8 @@ end
 
 function get_Δw(param_set, w_up::FT, w_en::FT) where {FT}
```
Comment (odunbar) [inline src/closures/entr_detr.jl:26]: `param_set` type here?

File: src/closures/mixing_length.jl [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-    c_d::FT = CPEDMF.c_d(param_set)
-    smin_ub::FT = CPEDMF.smin_ub(param_set)
-    smin_rm::FT = CPEDMF.smin_rm(param_set)
-    l_max::FT = ICP.l_max(param_set)
-    c_b::FT = ICP.static_stab_coeff(param_set)
-    g::FT = CPP.grav(param_set)
-    molmass_ratio::FT = CPP.molmass_ratio(param_set)
-    vkc::FT = CPSGS.von_karman_const(param_set)
+function mixing_length(param_set::EDMFPS, ml_model::MinDisspLen{FT}) where {FT}
+    c_m::FT = param_set.c_m
```
Comment (odunbar) [inline src/closures/mixing_length.jl:2]: Our `param_set` object automatically types all the parameters into `FT` thus you should remove all `::FT` here.

File: src/closures/nondimensional_exchange_functions.jl [hunk 5]
```diff
@@ -1,8 +1,8 @@
 #### Non-dimensional Entrainment-Detrainment functions
 function max_area_limiter(param_set, max_area, a_up)
     FT = eltype(a_up)
```
Comment (odunbar) [inline src/closures/nondimensional_exchange_functions.jl:3]: Again, we already type our parameters in `param_set`. We should be able to remove all the `FT(param_set)` and just have `param_set`.

File: src/closures/nondimensional_exchange_functions.jl [hunk 6]
```diff
@@ -26,25 +26,25 @@ function non_dimensional_groups(param_set, εδ_model_vars)
...
 """
-function non_dimensional_function(param_set, εδ_model_vars, ::MDEntr)
+function non_dimensional_function(param_set::MDEntrainmentParameters, εδ_model_vars, ::MDEntr)
     FT = eltype(εδ_model_vars.q_cond_up)
     Δw = get_Δw(param_set, εδ_model_vars.w_up, εδ_model_vars.w_en)
-    c_ε = FT(CPEDMF.c_ε(param_set))
-    μ_0 = FT(CPEDMF.μ_0(param_set))
-    β = FT(CPEDMF.β(param_set))
-    χ = FT(CPEDMF.χ(param_set))
+    c_ε = FT(param_set.c_ε)
```
Comment (odunbar) [inline src/closures/nondimensional_exchange_functions.jl:40]: again can remove the `FT(...)`

File: src/EDMF_functions.jl [hunk 7]
```diff
@@ -396,13 +396,13 @@ function compute_updraft_top(grid::Grid{FT}, state::State, i::Int)::FT where {FT
...
-function compute_plume_scale_height(grid::Grid{FT}, state::State, param_set::APS, i::Int)::FT where {FT}
-    H_up_min::FT = CPEDMF.H_up_min(param_set)
+function compute_plume_scale_height(grid::Grid{FT}, state::State, param_set::EDMFPS, i::Int)::FT where {FT}
+    H_up_min::FT = param_set.H_up_min
     updraft_top = compute_updraft_top(grid, state, i)
     return max(updraft_top, H_up_min)
 end
 
-function compute_up_stoch_tendencies!(edmf::EDMFModel, grid::Grid, state::State, param_set::APS, surf::SurfaceBase)
+function compute_up_stoch_tendencies!(edmf::EDMFModel, grid::Grid, state::State, param_set::EDMFPS, surf::SurfaceBase)
```
Comment (odunbar) [inline src/EDMF_functions.jl:405]: This should be `param_set::AbstractStochasticEntrainmentClosureParameters` (or some abbreviation)?

**Review by odunbar — COMMENTED** [review]
Hi Costa - this looks great! 
3 points.
1) As we now do our type enforcement when we get parameter values out of the toml struct, we no longer should need any of these `alpha::FT` or `FT(alpha(param_set))` type expressions. These can be removed.
2) For now i think it will be good practice to always give `param_set` a type: some are missing in e.g. `entr_detr.jl` function definitions. 
3) In theory we should have **no** calls to something like `param_set.box.parameter` (only ever `param_set.parameter`). Where these appear - this indicates that either the dispatching is off (e.g. the code should wrap the code in a new function), or maybe the parameter needs to be put in `param_set` - we can deal with these in a case-by-case basis offline.

**Comment by charleskawczynski** [comment]
Can we first get [CP 62](https://github.com/CliMA/CLIMAParameters.jl/pull/62) in mergeable shape?

**Comment by charleskawczynski** [comment]
@trontrytel is working on internalizing / isolating TC's ClimaParameters in #1000, so we should be able to revert _all_ of the source code changes, and only focus on the parameter boxes in this PR.

**Comment by odunbar** [comment]
@charleskawczynski The internal climaparameters do not actually help us in the microphysics case.

They are a method of going from a Hierarchical storage of CP (currently in CP) to a flat box in the code source (ICP).

The new climaparameters interface is the opposite, going from flat storage of CP (toml files) to a hierarchical deployment in the code source (Parameter boxes).

**Comment by charleskawczynski** [comment]
Superseded by #1192.

**Pull request closed without merging**